internal/app/handler/handler.go

```diff
@@ -0,0 +1,59 @@
+package handler
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/SAgamyradov/yandexService.git/internal/app/repository"
+	"github.com/gin-gonic/gin"
+)
+
+// POST request
+func ShortenURL(c *gin.Context, repo repository.URLRepository, baseURL string) {
+	if c.Request.Method != http.MethodPost {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Incorrect method"})
+		return
+	}
+
+	body, err := io.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Error reading"})
+		return
+	}
+
+	longURL := strings.TrimSpace(string(body))
+	if _, err = url.ParseRequestURI(longURL); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Incorrect URL"})
+		return
+	}
+
+	shortURLId, err := repo.GenerateShortURL(longURL)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Error saving URL"})
+		return
+	}
+
+	shortURL := baseURL + shortURLId
```

Comment:
[CHANGE REQUEST] Please avoid business logic in the presentation layer
https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6#:~:text=The%20presentation%20layer%20is%20the,Update%20or%20Delete)%20operation%20flow